packages/fxa-content-server/server/lib/routes/payment.js

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const flowMetrics = require('../flow-metrics');
+const logger = require('../logging/log')('routes.index');
+const vhost = require('vhost');
+
+module.exports = function (config) {
+  let featureFlags;
+  const featureFlagConfig = config.get('featureFlags');
+  if (featureFlagConfig.enabled) {
+    featureFlags = require('fxa-shared/feature-flags')(featureFlagConfig, logger);
+  } else {
+    featureFlags = { get: () => ({}) };
+  }
+
+  const ENV = config.get('env');
+  const FLOW_ID_KEY = config.get('flow_id_key');
+  const PAYMENT_URL = config.get('payment_url');
+  const PAYMENT_HOST = PAYMENT_URL && (new URL(PAYMENT_URL)).host;
+  const PROFILE_SERVER_URL = config.get('profile_url');
+  const STATIC_RESOURCE_URL = config.get('static_resource_url');
+  // add version from package.json to config
+  const RELEASE = require('../../../package.json').version;
+  const WEBPACK_PUBLIC_PATH = `${STATIC_RESOURCE_URL}/${config.get('jsResourcePath')}/`;
+
+  const serializedConfig = encodeURIComponent(JSON.stringify({
+    env: ENV,
+    profileUrl: PROFILE_SERVER_URL,
+    release: RELEASE,
+    staticResourceUrl: STATIC_RESOURCE_URL,
+    webpackPublicPath: WEBPACK_PUBLIC_PATH,
+  }));
+
+  return {
+    method: 'get',
+    path: '/payment',
+    process: async function (req, res) {
+      const host = req.headers.host;
+      if (host !== PAYMENT_HOST) {
```

Comment:
This is not an approved way to do secure per-host restrictions ;)

But the fragility of this kind of restriction does make it seem like there should be two servers.